### PR TITLE
support 4d softmax

### PR DESCRIPTION
**Pull request by dding3**
This PR is split from https://github.com/intel-analytics/analytics-zoo/pull/966/files and used to fix https://github.com/intel-analytics/analytics-zoo/issues/961

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.zoo.pipeline.api.keras.layers
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.nn.keras.{KerasLayer, SoftMax => KSoftMax}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.{MultiShape, Shape, SingleShape}
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala]: why need import Activity, KerasLayer, Multi/SingleShape?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    if (input.length <= 2) {
+      layer.asInstanceOf[AbstractModule[Tensor[T], Tensor[T], T]]
+    }
+    else {
+      val seq = TSequential[T]()
+      seq.add(Transpose(Array((1, 3))))
+      seq.add(layer)
+      seq.add(Transpose(Array((1, 3))))
+
+      val model = if (input.length > 3) {
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala]: is input >=5D also supported in this case?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala (hunk 1 above)
Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala]: Removed the unnecessary import.

**Comment by dding3** [comment]
Jekins passed. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1888/

Thanks for your comments.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala (hunk 2 above)
Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/Softmax.scala]: Updated
